runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -131,14 +133,14 @@ public Token appendCheckpoint() {
         // Queries the sequencer for the global log tail. We then read the global log tail to
         // persist the entry on the logunit in turn preventing from a new sequencer from
         // regressing tokens.
-        Token markerToken = rt.getSequencerView().query().getToken();
-        if (Address.nonAddress(markerToken.getSequence())) {
-            return markerToken;
+        TokenResponse tokenResponse = rt.getSequencerView().next();
```

Comment:
Some tests might be failing due to this change too. Tests assume no token is reserved for trimming.
Is there a specific reason for this change?

Reply:
Nope, its not because of that. Tests are failing because of the `MultiCheckpointWriter.java` change

I think when I did this change, we were just take the current tail (not a new token), so it was possible to generate a hole on a stream, which is not good.